test/test_train_mp_imagenet_amp.py

```diff
@@ -219,7 +219,13 @@ def train_imagenet():
       summary_writer=writer)
   loss_fn = nn.CrossEntropyLoss()
   if FLAGS.amp:
-    scaler = GradScaler(use_zero_grad=FLAGS.use_zero_grad)
+    if device_hw == 'TPU':
+        autocast = torch.xla.amp.autocast
+        scaler = None
+    elif device_hw == 'GPU':
+        autocast = torch.cuda.amp.autocast
+        # GradScaler only used for GPU
+        scaler = GradScaler(use_zero_grad=FLAGS.use_zero_grad)
```

Comment:
Why don't we use a scaler for TPU?

Reply:
That comment was a bit misleading. GradScaler is necessary for float16 loss but not bfloat16. float16 has less dynamic range than float32 since it uses only 5 exponent bits vs 8 exponent bits, so the loss needs to be scaled to prevent small values from disappearing. bfloat16 has the same dynamic range as float32 so doesn't have this problem.

Reply:
That means TPU uses bfloat16 by default?

Reply:
For AMP, yes!